### PR TITLE
[site,doc] Restyle the pagetoc

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
After #18411 was merged, I think the styling left a bit to be desired.
This PR restyles the sidebar ToC to be more minimalist. It also adds better highlighting behaviour to show context when you hover.

---

DEMO : [staging.opentitan.org](https://staging.opentitan.org/book/hw/ip/otbn/dv/doc/fcov.html)

---
image

### CODE REVIEW

**Review by crteja — APPROVED** [review]
Thanks for the changes @hcallahan-lowrisc . This looks much better and easier to navigate

**Comment by andreaskurth** [comment]
Nice @hcallahan-lowrisc :+1: I especially like that it's less obtrusive than the previous design. I find the font relatively hard to read, though. Removing `font-family: "Recursive", sans-serif` makes it much more pleasing to my eye.

File: site/book-theme/pagetoc.css [hunk 1]
```diff
@@ -15,28 +15,33 @@
...
+        width: calc(40% - 15vw + 100px);
         max-width: 400px;
         top: calc(var(--menu-bar-height) + 8rem);
         scroll-behavior: smooth;
     }
     .pagetoc {
-        font-size: 95%;
+        font-family: 'Recursive', sans-serif;
+        font-size: 90%;
+        font-weight: 300;
```
Comment (jwnrt) [inline site/book-theme/pagetoc.css:26]: I prefer the text without these three settings: it's quite hard to read with the other font and when it's thin.

**Review by jwnrt — COMMENTED** [review]
Thanks Harry, I agree this is nicer and less obstrusive. My only comment is on the text readability, otherwise great

**Comment by hcallahan-lowrisc** [comment]
Interesting, thanks for the reviews. I actually prefer the different font to contrast the main body of text, but agree that it's slightly harder to read. I'll wait for others to comment, but will try some other fonts.
One reason for choosing that font in particular is that is a multiplexed/uniwidth font, which means that the width of each character does not change when changing the weight, e.g. light, regular, bold. With the default font, if you make a heading bold it becomes much wider, which doesn't look so great as the heading jump about when you hover, especially with longer headings.

**Comment by hcallahan-lowrisc** [comment]
After looking at this on James' laptop, the font clearly looked different to how it was rendered on my own, it was much harder to read.. More investigation needed.

**Review by HU90m — APPROVED** [review]
I like the look of this TOC even more, thanks Harry.

Not sure about using a different font for the TOC, but I guess it's a cool font so why not.

**Comment by hcallahan-lowrisc** [comment]
I've tweaked the styling again. Still the same font but a heavier weight, this time with a grey colour so the bold, hovered headings contrast against the others. Updated the demo site with these changes. LMKWYT. Thanks

**Review by andreaskurth — APPROVED** [review]
LGTM based on the current demo. Thanks for taking our feedback into account, @hcallahan-lowrisc :+1:

**Comment by jwnrt** [comment]
Thanks for the update @hcallahan-lowrisc - it's much more readable now.

I think I'd still prefer not introducing a font but understand it's because this one is a uniform width between normal and bold?

Would be good to serve the font locally too since I'm not sure connections to Google Fonts meet GDPR.

**Comment by hcallahan-lowrisc** [comment]
> Thanks for the update @hcallahan-lowrisc - it's much more readable now.
> 
> I think I'd still prefer not introducing a font but understand it's because this one is a uniform width between normal and bold?


Yeah the new font is mainly because of the uniwidth property, see above https://github.com/lowRISC/opentitan/pull/18841#issuecomment-1581109571.

> 
> Would be good to serve the font locally too since I'm not sure connections to Google Fonts meet GDPR.

I had never heard of this issue before, thanks for pointing it out! Switched to serving the font-file locally.

**Pull request merged**